Jenkinsfile

```diff
@@ -490,6 +532,115 @@ az aks create                      \
+id=\$(aws route53 list-hosted-zones-by-name --dns-name "${dnsZone}" --max-items 1 --query 'HostedZones[0].Id' --output text)
+aws route53 get-hosted-zone --id "\$id" --query DelegationSet.NameServers
+"""
+                                                )
+                                                for (ns in readJSON(text: output)) {
+                                                    nameServers << ns
+                                                }
+                                            }
+                                            insertGlueRecords(dnsPrefix, nameServers, "60", parentZone, parentZoneResourceGroup)
+                                        }
+                                    } finally {
+                                        // Delete EKS cluster
+                                        container('eksctl') {
+                                            sh "eksctl delete cluster --name ${clusterName} --timeout 60m"
+                                        }
+                                        container('aws') {
+                                            withEnv([
+                                                "PATH+JQ=${tool 'jq'}",
+                                                "HOME=${env.WORKSPACE}",
+                                            ]) {
+                                                sh """
+                                                    #!/bin/bash
+                                                    CONFIG="${clusterName}-autogen.json"
+                                                    ACCOUNT=\$(aws sts get-caller-identity --query Account --output text)
+                                                    aws iam detach-user-policy --user-name "bkpr-${dnsZone}" --policy-arn "arn:aws:iam::\${ACCOUNT}:policy/bkpr-${dnsZone}"
+                                                    aws iam delete-policy --policy-arn "arn:aws:iam::\${ACCOUNT}:policy/bkpr-${dnsZone}"
+                                                    ACCESS_KEY_ID=\$(cat \${CONFIG} | jq -r .externalDns.aws_access_key_id)
+                                                    aws iam delete-access-key --user-name "bkpr-${dnsZone}" --access-key-id "\${ACCESS_KEY_ID}"
+                                                    aws iam delete-user --user-name "bkpr-${dnsZone}"
+                                                    CLIENT_ID=\$(cat \${CONFIG} | jq -r .oauthProxy.client_id)
+                                                    aws cognito-idp delete-user-pool-client --user-pool-id "${awsUserPoolId}" --client-id "\${CLIENT_ID}"
+                                                    DNS_ZONE_ID=\$(aws route53 list-hosted-zones-by-name --dns-name "${dnsZone}" --max-items 1 --query 'HostedZones[0].Id' --output text)
+                                                    aws route53 list-resource-record-sets --hosted-zone-id \${DNS_ZONE_ID} | jq -c '.ResourceRecordSets[]' | while read -r resourcerecordset; do
```

Comment:
I haven't tested it in anger, but I think the following is the same fwiw.  (apparently `--query` can do all sorts of things!)

```sh
aws route53 list-resource-record-sets \
  --hosted-zone-id \${DNS_ZONE_ID} \
  --query '{ChangeBatch:{Changes:ResourceRecordSets[?Type != `NS` && Type != `SOA`].{Action:`DELETE`,ResourceRecordSet:@}}}' \
  --output json > changes

aws route53 change-resource-record-sets \
  --cli-input-json file://changes \
  --hosted-zone-id \${DNS_ZONE_ID} \
  --query 'ChangeInfo.Id' \
  --output text

aws route53 delete-hosted-zone \
  --id \${DNS_ZONE_ID} \
  --query 'ChangeInfo.Id' \
  --output text
```

Reply:
Works!